docs/src/pages/components/data-grid/columns/ColumnSelectorGrid.js

```diff
@@ -11,7 +11,12 @@ export default function ColumnSelectorGrid() {
 
   return (
     <div style={{ height: 400, width: '100%' }}>
-      <DataGrid {...data} showToolbar />
+      <DataGrid
+        {...data}
+        components={{
+          Header: GridHeader,
```

Comment:
if the user does not add that then he won't be able to open the columns or filter panel in the current state

Reply:
Another reason to move forward with https://github.com/mui-org/material-ui-x/issues/946#issuecomment-769758055 :)

Reply:
Ok, so that is because of the `PreferencesPanel`. Then I agree we should have it by default because if not then we need to remove those functionalities from the column menu as well.

Reply:
I would propose the following:

- We pause this PR, work on #946 first.
- We add a `PreferencesPanel` slot.
- We render the PreferencesPanel slot inside the main grid, it doesn't need to be inside the header.
- We expose the PreferencesPanel module as `GridPreferencesPanel`
- We resume the work on this PR, after rebasing.

Reply:
I'm ok with the approach.

Reply:
cc @dtassone